internal/checks/blockheight.go

```diff
@@ -122,13 +133,15 @@ func (c *BlockHeightCheck) GetError() error {
 func (c *BlockHeightCheck) subscribeNewHead() error {
 	onNewHead := func(header *types.Header) {
 		c.blockHeight = header.Number.Uint64()
+		metrics.BlockHeight.WithLabelValues(c.upstreamConfig.ID, c.upstreamConfig.HTTPURL).Set(float64(c.blockHeight))
 		c.webSocketError = nil
 		c.blockHeightError = nil
 	}
 
 	onError := func(failure string) {
 		zap.L().Error("Encountered error in NewHead Websockets subscription.", zap.Any("upstreamID", c.upstreamConfig.ID), zap.String("WSURL", c.upstreamConfig.WSURL))
 
+		metrics.BlockHeightErrors.WithLabelValues(c.upstreamConfig.ID, c.upstreamConfig.HTTPURL).Inc()
```

Comment:
Would it be helpful to label the errors? There are many different kinds for block height checking and some are more meaningful than others.

Reply:
yeah, look at the interfaces it looks tricky to do that atm.

The error returned is a generic `error` type - the only type of information that error provides is an error string - can you put those as labels?

Reply:
In our logic I suppose the types of failures are WS ones — initialization failure is one, and the other is an error encountered after successful init. And then there's HTTP failure

Reply:
Errors strings as labels aren't ideal b/c they can be very high cardinality and hard to build graphs off of. I was thinking we could have string enums for categories of errors, such as `get_client_failure`, `http_get_Head_failure`, `ws_subscribe_new_head_failure`, etc and not break down errors within each category.

Reply:
gotcha! made some changes